segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/ServiceStarter.java

```diff
@@ -166,18 +171,22 @@ private void attachStorage(ServiceBuilder builder) {
                 switch (storageChoice) {
                     case HDFS:
                         HDFSStorageConfig hdfsConfig = setup.getConfig(HDFSStorageConfig::builder);
-                        return new HDFSStorageFactory(hdfsConfig, setup.getExecutor());
+                        StorageMetricsBase metrics = new HDFSMetrics();
```

Comment:
Since you don't seem to use this metrics anywhere, can we initialize them inside the constructors for the factories? That will avoid the need for an extra parameter and hence you won't need to touch so many files.